app/stagemanager.js

```diff
@@ -86,26 +86,35 @@ class StageManager extends MyTrello {
     }
 
     closeUnusedStages(data) {
-      var deferred = Q.defer();
-        var stages = data[0].map(value => value['name']),
-            self = this;
+      var deferred = Q.defer(),
+          stages = data[0].map(value => value['name']),
+          closing = [];
 
         data[1].forEach(function(trelloList) {
+          var listDefer = Q.defer();
+          closing.push(listDefer.promise);
             if (!(stages.includes(trelloList.name))) {
```

Comment:
If this conditional fails, then `listDefer` will never resolve or reject.  This could result in the main promise for the method also never resolving/rejecting.